controllers/flp/flp_controller.go

```diff
@@ -253,3 +266,84 @@ func reconcileLokiRoles(ctx context.Context, r *reconcilers.Common, b *builder)
 	}
 	return nil
 }
+
+func (r *Reconciler) getOpenShiftSubnets(ctx context.Context) ([]flowslatest.SubnetLabel, error) {
+	var subnets []flowslatest.SubnetLabel
+
+	// Pods and Services subnets are found in CNO config
+	if r.mgr.HasCNO() {
+		network := &operatorv1.Network{}
+		err := r.Get(ctx, types.NamespacedName{Name: "cluster"}, network)
+		if err != nil {
+			return nil, fmt.Errorf("can't get Network information: %w", err)
+		}
+		var podCIDRs []string
+		for _, podsNet := range network.Spec.ClusterNetwork {
+			podCIDRs = append(podCIDRs, podsNet.CIDR)
+		}
+		if len(podCIDRs) > 0 {
+			subnets = append(subnets, flowslatest.SubnetLabel{
+				Name:  "Pods",
+				CIDRs: podCIDRs,
+			})
+		}
+		svcCIDRs := network.Spec.ServiceNetwork
```

Comment:
u need to loop over `serviceNetwork` cidrs similar to what u did for `clusterNetwork`

Reply:
it seems there is a part for external service in https://docs.openshift.com/container-platform/4.14/rest_api/config_apis/network-config-openshift-io-v1.html#spec-externalip

Reply:
I haven't implemented it yet, asking some details to the SDN team bc it's in a different API object and I'm not sure what can be considered the source of truth